File: scenarios/moderate/cross-platform/qt/n3.cpp
#include <QMessageBox>
#include <QApplication>
    : QWidget(parent)
    , m_isModalOpen(false)
{
    setupUI();
}
{
    // Main layout
    QVBoxLayout *mainLayout = new QVBoxLayout(this);
    mainLayout->setSpacing(20);
    mainLayout->setContentsMargins(20, 20, 20, 20);
    // Main content card
    QFrame *mainCard = new QFrame();
    mainCard->setFrameStyle(QFrame::Box | QFrame::Raised);
    mainCard->setLineWidth(1);
    mainCard->setStyleSheet(
        "QFrame {"
        "    background-color: #f8f9fa;"
        "    border: 1px solid #dee2e6;"
        "    border-radius: 8px;"
        "    padding: 20px;"
        "}"
    );
    QVBoxLayout *cardLayout = new QVBoxLayout(mainCard);
    cardLayout->setSpacing(20);
    // Title
        "    font-size: 24px;"
        "    font-weight: bold;"
        "    color: #333333;"
        "    margin-bottom: 10px;"
        "}"
    );
    // Description
        "    color: #666666;"
        "    font-size: 16px;"
        "    margin-bottom: 20px;"
        "}"
    );
    // Open Modal Button
    m_openModalButton = new QPushButton("Open Settings Modal");
    m_openModalButton->setStyleSheet(
        "QPushButton {"
        "    background-color: #007bff;"
        "    color: white;"
        "    border: none;"
        "    border-radius: 4px;"
        "    padding: 15px 30px;"
        "    font-size: 16px;"
        "    font-weight: bold;"
        "}"
        "QPushButton:hover {"
        "    background-color: #0056b3;"
        "}"
    );
    cardLayout->addWidget(m_openModalButton);
    // Description
    );
        "    color: #666666;"
        "    font-size: 14px;"
        "    margin-top: 20px;"
        "}"
    );
    mainLayout->addWidget(mainCard);
    QFrame *infoFrame = new QFrame();
    infoFrame->setFrameStyle(QFrame::Box | QFrame::Raised);
    infoFrame->setLineWidth(1);
    infoFrame->setStyleSheet(
        "QFrame {"
        "    background-color: #e9ecef;"
        "    border: 1px solid #dee2e6;"
        "    border-radius: 4px;"
        "    padding: 20px;"
        "}"
    );
    QVBoxLayout *infoLayout = new QVBoxLayout(infoFrame);
    infoLayout->setSpacing(15);
        "• Escape key not handled: Modal doesn't close when Escape is pressed"
    );
    issuesText->setStyleSheet(
        "    color: #666666;"
        "    font-size: 12px;"
        "}"
    );
    issuesText->setWordWrap(true);
    infoLayout->addWidget(issuesText);
    // How to Fix
    fixTitle->setStyleSheet(
        "    font-size: 16px;"
        "    font-weight: bold;"
        "    color: #333333;"
        "}"
    );
    infoLayout->addWidget(fixTitle);
        "• Handle Escape key to close modal"
    );
    fixText->setStyleSheet(
        "    color: #666666;"
        "    font-size: 12px;"
        "}"
    );
    fixText->setWordWrap(true);
    infoLayout->addWidget(fixText);
    mainLayout->addWidget(infoFrame);
}
{
    // Basic setup only
    m_openModalButton->setAccessibleName("Open Settings Modal button");
}
{
    if (m_isModalOpen) return;
    // Show modal
    showModal();
}
{
    m_modalDialog = new QDialog(this);
    m_modalDialog->setWindowTitle("Settings");
    m_modalDialog->setModal(true);
    m_modalDialog->setFixedSize(500, 400);
    m_modalDialog->setWindowFlags(Qt::Dialog | Qt::WindowTitleHint | Qt::WindowCloseButtonHint);
    // Modal layout
    QVBoxLayout *modalLayout = new QVBoxLayout(m_modalDialog);
    modalLayout->setSpacing(20);
    modalLayout->setContentsMargins(30, 30, 30, 30);
    // Modal Header
    QHBoxLayout *headerLayout = new QHBoxLayout();
        "    font-size: 20px;"
        "    font-weight: bold;"
        "    color: #333333;"
        "}"
    );
    headerLayout->addStretch();
    m_closeModalButton = new QPushButton("×");
    m_closeModalButton->setFixedSize(30, 30);
    m_closeModalButton->setStyleSheet(
        "QPushButton {"
        "    background-color: transparent;"
        "    border: none;"
        "    color: #666666;"
        "    font-size: 16px;"
        "    font-weight: bold;"
        "}"
        "QPushButton:hover {"
        "    background-color: #e9ecef;"
        "    border-radius: 15px;"
        "}"
    );
    headerLayout->addWidget(m_closeModalButton);
    modalLayout->addLayout(headerLayout);
    // Settings Form
    QFormLayout *settingsLayout = new QFormLayout();
    // Theme Setting
        "    font-weight: bold;"
        "    color: #333333;"
        "}"
    );
    m_themeCombo = new QComboBox();
    m_themeCombo->addItems({"Light", "Dark", "Auto"});
    m_themeCombo->setCurrentText("Light");
    m_themeCombo->setStyleSheet(
        "QComboBox {"
        "    padding: 10px;"
        "    border: 2px solid #ddd;"
        "    border-radius: 4px;"
        "    font-size: 14px;"
        "}"
    );
    // Language Setting
        "    font-weight: bold;"
        "    color: #333333;"
        "}"
    );
    m_languageCombo = new QComboBox();
    m_languageCombo->addItems({"English", "Spanish", "French"});
    m_languageCombo->setCurrentText("English");
    m_languageCombo->setStyleSheet(
        "QComboBox {"
        "    padding: 10px;"
        "    border: 2px solid #ddd;"
        "    border-radius: 4px;"
        "    font-size: 14px;"
        "}"
    );
    // Notifications Setting
        "    font-weight: bold;"
        "    color: #333333;"
        "}"
    );
    m_notificationsCombo = new QComboBox();
    m_notificationsCombo->addItems({"All", "Important Only", "None"});
    m_notificationsCombo->setCurrentText("All");
    m_notificationsCombo->setStyleSheet(
        "QComboBox {"
        "    padding: 10px;"
        "    border: 2px solid #ddd;"
        "    border-radius: 4px;"
        "    font-size: 14px;"
        "}"
    );
    modalLayout->addLayout(settingsLayout);
    modalLayout->addStretch();
    // Modal Actions
    QHBoxLayout *actionsLayout = new QHBoxLayout();
    actionsLayout->addStretch();
    m_cancelButton = new QPushButton("Cancel");
    m_cancelButton->setStyleSheet(
        "QPushButton {"
        "    background-color: #6c757d;"
        "    color: white;"
        "    border: none;"
        "    border-radius: 4px;"
        "    padding: 10px 20px;"
        "    font-size: 14px;"
        "    font-weight: bold;"
        "}"
        "QPushButton:hover {"
        "    background-color: #5a6268;"
        "}"
    );
    actionsLayout->addWidget(m_cancelButton);
    m_saveButton = new QPushButton("Save Settings");
    m_saveButton->setStyleSheet(
        "QPushButton {"
        "    background-color: #007bff;"
        "    color: white;"
        "    border: none;"
        "    border-radius: 4px;"
        "    padding: 10px 20px;"
        "    font-size: 14px;"
        "    font-weight: bold;"
        "}"
        "QPushButton:hover {"
        "    background-color: #0056b3;"
        "}"
    );
    actionsLayout->addWidget(m_saveButton);
    modalLayout->addLayout(actionsLayout);
    // Show modal
    m_isModalOpen = true;
    m_modalDialog->show();
    // Should handle Escape key to close modal
}
{
    closeModal();
}
{
    closeModal();
}
{
    // Simulate saving settings
    QMessageBox::information(m_modalDialog, "Success", "Settings saved successfully!");
    closeModal();
}
{
    if (!m_isModalOpen) return;
    // Hide modal
    m_modalDialog->close();
    m_modalDialog->deleteLater();
    m_modalDialog = nullptr;
    m_isModalOpen = false;
    } else {
    }
}
// Should implement methods to:

File: scenarios/moderate/desktop/linux-gtk/n3.py
#!/usr/bin/env python3
"""
"""
import gi
gi.require_version('Gtk', '3.0')
from gi.repository import Gtk, GObject, GLib
    def __init__(self):
        self.set_default_size(600, 700)
        self.set_resizable(True)
        # State
        self.is_modal_open = False
        self.setup_ui()
    def setup_ui(self):
        # Main container
        main_box = Gtk.Box(orientation=Gtk.Orientation.VERTICAL, spacing=20)
        main_box.set_margin_left(20)
        main_box.set_margin_right(20)
        main_box.set_margin_top(20)
        main_box.set_margin_bottom(20)
        self.add(main_box)
        # Main content card
        main_card = Gtk.Frame()
        main_card.set_shadow_type(Gtk.ShadowType.ETCHED_IN)
        main_card.set_margin_bottom(20)
        card_content = Gtk.Box(orientation=Gtk.Orientation.VERTICAL, spacing=20)
        card_content.set_margin_left(30)
        card_content.set_margin_right(30)
        card_content.set_margin_top(30)
        card_content.set_margin_bottom(30)
        main_card.add(card_content)
        # Title
        # Description
        # Open Modal Button
        self.open_modal_button.set_size_request(-1, 40)
        self.open_modal_button.connect("clicked", self.on_open_modal_clicked)
        card_content.pack_start(self.open_modal_button, False, False, 0)
        # Description
        desc_text.set_halign(Gtk.Align.START)
        desc_text.set_line_wrap(True)
        card_content.pack_start(desc_text, False, False, 0)
        main_box.pack_start(main_card, False, False, 0)
        # Modal Overlay
        self.modal_overlay = Gtk.Fixed()
        self.modal_overlay.set_visible(False)
        self.modal_overlay.override_background_color(Gtk.StateFlags.NORMAL, Gtk.RGBA(0, 0, 0, 0.5))
        main_box.pack_start(self.modal_overlay, True, True, 0)
        # Modal Dialog
        self.create_modal_dialog()
    def create_modal_dialog(self):
        # Modal container
        self.modal_dialog = Gtk.Frame()
        self.modal_dialog.set_size_request(500, 400)
        self.modal_dialog.set_shadow_type(Gtk.ShadowType.OUT)
        self.modal_dialog.set_halign(Gtk.Align.CENTER)
        self.modal_dialog.set_valign(Gtk.Align.CENTER)
        modal_content = Gtk.Box(orientation=Gtk.Orientation.VERTICAL, spacing=20)
        modal_content.set_margin_left(30)
        modal_content.set_margin_right(30)
        modal_content.set_margin_top(30)
        modal_content.set_margin_bottom(30)
        self.modal_dialog.add(modal_content)
        # Modal Header
        header_box = Gtk.Box(orientation=Gtk.Orientation.HORIZONTAL, spacing=0)
        modal_content.pack_start(header_box, False, False, 0)
        modal_title.set_markup("<span size='large' weight='bold'>Settings</span>")
        modal_title.set_halign(Gtk.Align.START)
        header_box.pack_start(modal_title, True, True, 0)
        self.close_modal_button.set_size_request(30, 30)
        self.close_modal_button.connect("clicked", self.on_close_modal_clicked)
        header_box.pack_start(self.close_modal_button, False, False, 0)
        # Settings Form
        settings_box = Gtk.Box(orientation=Gtk.Orientation.VERTICAL, spacing=20)
        modal_content.pack_start(settings_box, False, False, 0)
        # Theme Setting
        theme_box = Gtk.Box(orientation=Gtk.Orientation.VERTICAL, spacing=8)
        settings_box.pack_start(theme_box, False, False, 0)
        self.theme_combo = Gtk.ComboBoxText()
        self.theme_combo.append_text("Light")
        self.theme_combo.append_text("Dark")
        self.theme_combo.append_text("Auto")
        self.theme_combo.set_active(0)
        theme_box.pack_start(self.theme_combo, False, False, 0)
        # Language Setting
        language_box = Gtk.Box(orientation=Gtk.Orientation.VERTICAL, spacing=8)
        settings_box.pack_start(language_box, False, False, 0)
        self.language_combo = Gtk.ComboBoxText()
        self.language_combo.append_text("English")
        self.language_combo.append_text("Spanish")
        self.language_combo.append_text("French")
        self.language_combo.set_active(0)
        language_box.pack_start(self.language_combo, False, False, 0)
        # Notifications Setting
        notifications_box = Gtk.Box(orientation=Gtk.Orientation.VERTICAL, spacing=8)
        settings_box.pack_start(notifications_box, False, False, 0)
        self.notifications_combo = Gtk.ComboBoxText()
        self.notifications_combo.append_text("All")
        self.notifications_combo.append_text("Important Only")
        self.notifications_combo.append_text("None")
        self.notifications_combo.set_active(0)
        notifications_box.pack_start(self.notifications_combo, False, False, 0)
        # Modal Actions
        actions_box = Gtk.Box(orientation=Gtk.Orientation.HORIZONTAL, spacing=10)
        actions_box.set_halign(Gtk.Align.END)
        modal_content.pack_start(actions_box, False, False, 20)
        self.cancel_button.set_size_request(100, 32)
        self.cancel_button.connect("clicked", self.on_cancel_clicked)
        actions_box.pack_start(self.cancel_button, False, False, 0)
        self.save_button.set_size_request(100, 32)
        self.save_button.connect("clicked", self.on_save_clicked)
        actions_box.pack_start(self.save_button, False, False, 0)
        # Add modal to overlay
        self.modal_overlay.put(self.modal_dialog, 0, 0)
        info_frame = Gtk.Frame()
        info_frame.set_shadow_type(Gtk.ShadowType.ETCHED_IN)
        info_content = Gtk.Box(orientation=Gtk.Orientation.VERTICAL, spacing=15)
        info_content.set_margin_left(20)
        info_content.set_margin_right(20)
        info_content.set_margin_top(20)
        info_content.set_margin_bottom(20)
        info_frame.add(info_content)
• Escape key not handled: Modal doesn't close when Escape is pressed""")
        issues_text.set_halign(Gtk.Align.START)
        issues_text.set_line_wrap(True)
        info_content.pack_start(issues_text, False, False, 0)
        # How to Fix
        fix_title.set_markup("<b>How to Fix:</b>")
        fix_title.set_halign(Gtk.Align.START)
        info_content.pack_start(fix_title, False, False, 0)
• Handle Escape key to close modal""")
        fix_text.set_halign(Gtk.Align.START)
        fix_text.set_line_wrap(True)
        info_content.pack_start(fix_text, False, False, 0)
        parent_box.pack_start(info_frame, False, False, 0)
        # Basic setup only
        self.open_modal_button.set_name("open-modal-button")
        self.close_modal_button.set_name("close-modal-button")
        self.save_button.set_name("save-button")
        self.cancel_button.set_name("cancel-button")
    def on_open_modal_clicked(self, button):
        if self.is_modal_open:
            return
        # Show modal
        self.modal_overlay.set_visible(True)
        self.is_modal_open = True
        # Should handle Escape key to close modal
    def on_close_modal_clicked(self, button):
        self.close_modal()
    def on_cancel_clicked(self, button):
        self.close_modal()
    def on_save_clicked(self, button):
        # Simulate saving settings
        dialog = Gtk.MessageDialog(
            parent=self,
            flags=Gtk.DialogFlags.MODAL,
            type=Gtk.MessageType.INFO,
            buttons=Gtk.ButtonsType.OK,
            message_format="Settings saved successfully!"
        )
        dialog.run()
        dialog.destroy()
        self.close_modal()
    def close_modal(self):
        if not self.is_modal_open:
            return
        # Hide modal
        self.modal_overlay.set_visible(False)
        self.is_modal_open = False
        else:
    def do_key_press_event(self, event):
        # Should close modal when Escape key is pressed
        # Should only work when modal is open
        if event.keyval == Gtk.KEY_Escape and self.is_modal_open:
            self.close_modal()
            return True
        return super().do_key_press_event(event)
    # Should implement methods to:
def main():
    app = Gtk.Application()
    app.run()
if __name__ == '__main__':
    main()
